core/grid.ts

```diff
@@ -184,6 +185,23 @@ export class Grid {
     this.pattern.setAttribute('y', `${y}`);
   }
 
+  /**
+   * Given a coordinate, return the nearest coordinate aligned to the grid.
+   *
+   * @returns Object with .x and .y properties in workspace coordinates,
+   *   or null if no change.
```

Comment:
* Please document parameter `xy`.
* I think it's not necessary to explain what a Coordinate is.
* 
```suggestion
   * @param xy A workspace coordinate.
   * @returns Workspace coordinate of nearest grid point, or null if
   *     xy was already a grid point.
```